sdks/go/pkg/beam/options/gcpopts/options_test.go

```diff
@@ -0,0 +1,79 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcpopts
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestGetProjectFromFlagOrEnvironmentWithNoProjectSet(t *testing.T) {
+	setupFakeCredentialFile(t, "")
+	*Project = ""
+	projectId := GetProjectFromFlagOrEnvironment(nil)
+	if projectId != "" {
+		t.Fatalf("%q returned as project id, should be \"\"", projectId)
```

Comment:
It's best to identify the function, the relevant inputs (here there aren't any, unless one wants to repeat the case information), what we wanted, and what we got.

So here, (and below), when possible, you can use a similar format.
```
if got, want := GetProjectFromFlagOrEnvironment(nil), ""; got != want {
   t.Fatalf("GetProjectFromFlagOrEnvironment() = %q, want %q", got, want)
}
```
It clearly indicates the call being run, and avoids repeating what we want and expect, and follows the "got, want" convention (eliding the got, since it's clear from the presentation that it's the output of the function). I have a personal preference to lean on defining both got and want within the if scope, so I don't make copy paste errors, when making various checks.

See https://github.com/golang/go/wiki/TestComments#got-before-want and the following sections. Overall it's a pretty good doc.